sfputil/main.py

```diff
@@ -687,123 +693,247 @@ def eeprom(port, dump_dom, namespace):
+                    pages = copy.deepcopy(SFF8472_MODULE_PAGES)
+                else:
+                    pages = [0]
+            else:
+                pages = copy.deepcopy(SFF8472_MODULE_PAGES) if not api.is_copper() else [0]
+                if page not in pages:
+                    pages.append(page)
+            return eeprom_hexdump_pages_sff8472(logical_port_name, pages, page)
         else:
-            output = eeprom_hexdump_sff8636(port, physical_port, page)
+            return ERROR_NOT_IMPLEMENTED, 'Cable type is not supported'
+    except NotImplementedError:
+        return ERROR_NOT_IMPLEMENTED, 'Sfp.read_eeprom() is currently not implemented for this platform'
 
-    output += '\n'
 
-    click.echo(output)
-
-def eeprom_hexdump_sff8472(port, physical_port, page):
-    try:
-        output = ""
-        indent = ' ' * 8
-        output += 'EEPROM hexdump for port {} page {}h'.format(port, page)
-        output += '\n{}A0h dump'.format(indent)
-        page_dump = platform_chassis.get_sfp(physical_port).read_eeprom(0, SFF8472_A0_SIZE)
-        if page_dump is None:
-            click.echo("Error: Failed to read EEPROM for A0h!")
-            sys.exit(ERROR_NOT_IMPLEMENTED)
+def eeprom_hexdump_pages_general(logical_port_name, pages, target_page):
+    """
+    Dump module EEPROM for given pages in hex format. This function is designed for cable type other than SFF8472.
+    Args:
+        logical_port_name: logical port name
+        pages: a list of pages to be dumped. The list always include a default page list and the target_page input by
+               user
+        target_page: user input page number, optional. target_page is only for display purpose
 
-        output += hexdump(indent, page_dump, 0)
-        page_dump = platform_chassis.get_sfp(physical_port).read_eeprom(SFF8472_A0_SIZE, PAGE_SIZE)
-        if page_dump is None:
-            click.echo("Error: Failed to read EEPROM for A2h!")
-            sys.exit(ERROR_NOT_IMPLEMENTED)
+    Returns:
+        tuple(0, dump string) if success else tuple(error_code, error_message)
+    """
+    if target_page is not None:
+        lines = [f'EEPROM hexdump for port {logical_port_name} page {target_page}h']
+    else:
+        lines = [f'EEPROM hexdump for port {logical_port_name}']
+    physical_port = logical_port_to_physical_port_index(logical_port_name)
+    for page in pages:
+        if page == 0:
+            lines.append(f'{EEPROM_DUMP_INDENT}Lower page 0h')
+            return_code, output = eeprom_dump_general(physical_port, page, 0, PAGE_SIZE, 0)
+            if return_code != 0:
+                return return_code, output
+            lines.append(output)
+
+            lines.append(f'\n{EEPROM_DUMP_INDENT}Upper page 0h')
+            return_code, output = eeprom_dump_general(physical_port, page, PAGE_OFFSET, PAGE_SIZE, PAGE_OFFSET)
+            if return_code != 0:
+                return return_code, output
+            lines.append(output)
```

Comment:
@Junchao-Mellanox can we avoid displaying upper page 0 if user input target page >= 1

Reply:
For backward compatible, previous implementation always print page 0. As you can see, I did not change the previous unit test cases, and all passed.